docs/source/spark.rst

```diff
@@ -0,0 +1,164 @@
+Spark development team is now focusing more on binary and native data formats
+with their new effort, Tungsten.
+
+Dask is written in Python, a multi-paradigm language built on top of the
+C/Fortan native stack.  This stack benefits from decades of scientific research
+optimizing very fast computation on numeric data.  As such, dask is already
+very good on analytic computations on data such as you might find in HDF5 files
+or analytic databases.  It can also handle JSON blob type data using Python
+data structures (which are `surprisingly fast`_) using the cytoolz_ library in
+parallel.
+
+
+Java Python Disconnect
+~~~~~~~~~~~~~~~~~~~~~~
+
+Python users on Spark sometimes express frustration by how far separated they
+are from computations.  Some of this is inevitable, distributed debugging is a
+hard problem.  Some of it however is due to having to hop over the JVM.  Spark
+workers spin up JVMs which in turn spin up Python processes.  Data moving back
+and forth makes extra trips both through a distributed cluster and also through
+extra serialization layers (see py4j_) and computation layers.  Limitations
+like the Java heap size and large Java stack traces come as a surprise to users
+accustomed to native code execution.
+
+Dask has an advantage for Python users because it is itself a Python library,
+so serialization and debugging when things go wrong happens more smoothly.
+
+However, dask has no benefit to non-Python users while Spark is useful in a
+variety of JVM languages (Scala, Java, Clojure) as well as limited support in
+Python and R.  New Spark projects like the DataFrame skip serialization and
+boxed execution issues by forgoing the Python process entirely and instead have
+Python code drive native Scala code.  APIs for these libraries tend to lag a
+bit behind their Scala counterparts.
+
+
+Scope
+~~~~~
+
+Spark was originally built around the RDD, an unordered collection allowing
+repeats.  Most spark add-ons were built on top of this construct, inheriting
+both its abilities and limitations.
+
+Dask is built on a lower-level and more general construct of a generic task
+graph with data dependencies.  This allows more general computations to be
+built by users within the dask framework.  This is probably the largest
+fundamental difference between the two projects.  Dask gives up high-level
+understanding to allow users to express more complex parallel algorithms.  This
+ended up being essential when writing complex projects like ``dask.array``.
+
+
+Developer-Facing Differences
+----------------------------
+
+Graph Granularity
+~~~~~~~~~~~~~~~~~
+
+Both Spark and Dask represent computations with directed acyclic graphs.  These
+graphs however represent computations at very different granularities.
+
+One operation on a Spark RDD might add a node like ``Map`` and ``Filter`` to
+the graph.  These are high-level operations that convey meaning and will
+eventually be turned into many little tasks to execute on individual nodes.
```

Comment:
Node here (node in a cluster) is an unfortunate homophone with node above (node in a graph). We could say vertex instead of node for the graph case, but that sounds more esoteric. Either way its no big deal.